components/eamxx/src/physics/cosp/cosp_c2f.F90

```diff
@@ -292,23 +292,44 @@ subroutine cosp_c2f_run(npoints, ncolumns, nlevels, ntau, nctp, &
  
     nptsperit = npoints
 
-    tca(:npoints,:nlevels) = cldfrac(:npoints,:nlevels)
-    cca(:npoints,:nlevels) = 0
-    mr_lsliq(:npoints,:nlevels) = 0
-    mr_ccliq(:npoints,:nlevels) = 0
-    mr_lsice(:npoints,:nlevels) = 0
-    mr_ccice(:npoints,:nlevels) = 0
-    dtau_c(:npoints,:nlevels) = 0
-    dtau_s(:npoints,:nlevels) = dtau067(:npoints,:nlevels)
-    dem_c (:npoints,:nlevels) = 0
-    dem_s (:npoints,:nlevels) = 1._wp - exp(-dtau105(:npoints,:nlevels))
-    mr_lsrain(:npoints,:nlevels) = 0
-    mr_ccrain(:npoints,:nlevels) = 0
-    mr_lssnow(:npoints,:nlevels) = 0
-    mr_lssnow(:npoints,:nlevels) = 0
-    mr_ccsnow(:npoints,:nlevels) = 0
-    mr_lsgrpl(:npoints,:nlevels) = 0
-    reff = 0  ! FIXME
+    ! In-cloud values are assumed. If ncolumns = 1, then convert in-cloud values to gridbox
+    if (ncolumns == 1) then
+       tca(:npoints,:nlevels) = cldfrac(:npoints,:nlevels)
+       cca(:npoints,:nlevels) = 0
+       mr_lsliq(:npoints,:nlevels) = 0
+       mr_ccliq(:npoints,:nlevels) = 0
+       mr_lsice(:npoints,:nlevels) = 0
+       mr_ccice(:npoints,:nlevels) = 0
+       dtau_c(:npoints,:nlevels) = 0
+       dtau_s(:npoints,:nlevels) = cldfrac(:npoints,:nlevels) * dtau067(:npoints,:nlevels)
+       dem_c (:npoints,:nlevels) = 0
+       dem_s (:npoints,:nlevels) = 1._wp - exp(-cldfrac(:npoints,:nlevels) * dtau105(:npoints,:nlevels))
+       mr_lsrain(:npoints,:nlevels) = 0
+       mr_ccrain(:npoints,:nlevels) = 0
+       mr_lssnow(:npoints,:nlevels) = 0
+       mr_lssnow(:npoints,:nlevels) = 0
+       mr_ccsnow(:npoints,:nlevels) = 0
+       mr_lsgrpl(:npoints,:nlevels) = 0
+       reff = 0  ! FIXME
```

Comment:
I know it was like this in the original, so no big deal, but what does this `FIXME` refer to? Is it something we should be concerned about? Should we beef up the comment, so that other devs may understand what needs to be fixed?

Reply:
I can add more context here. This will need to be fixed when we turn on the MODIS, CloudSat, and CALIPSO simulators I believe, but is not used for the ISCCP and MISR simulators.

Reply:
Okay I lied, I just decided to add doc about all these fields. Updated.